## 1. Import Library, Read each Dataset Files from the Merged of 5 Debate Videos from 3 Different YouTube Channels (Najwa Shihab, KOMPASTV, and KPU RI) and Combine them into 1 Centralized Dataset File

In [59]:
"""
Meng-import beberapa library yang dibutuhkan dalam penelitian kali ini
khususnya library/modul yang bertujuan untuk melakukan text processing seperti
text cleaning (membersihkan teks komentar dari simbol/karakter spesial), membersihkan setiap
teks komentar yang memiliki kalimat imbuhan (stopwords) atau kata yang tidak memiliki makna/arti,
mengubah struktur suatu kalimat utuh menjadi bagian-bagian yang lebih kecil (tokenization), dll
"""
import pandas as pd
from textblob import TextBlob
import re
import nltk
nltk.download("punkt")
from nltk.tokenize import word_tokenize
nltk.download("stopwords")
from nltk.corpus import stopwords

[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\ekapr\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\ekapr\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


### 1.1. Read/Load Dataset from Scraping Result based on Youtube Channel Najwa Shihab

In [60]:
"""
Membaca file dataset dari hasil scraping komentar-komentar pada video debat babak
pertama hingga babak kelima capres-cawapres dari channel YouTube Najwa Shihab
"""
data_najwa_shihab = pd.read_excel("Dataset/Dataset Komentar Video Debat dari YouTube Channel Najwa Shihab.xlsx")


"""
Menampilkan 8 data teratas dari variabel DataFrame "data_najwa_shihab"
yang memiliki jumlah total baris sebanyak 22.999 data yang didapatkan
dari video komentar debat babak pertama hingga babak kelima capres-cawapres di channel YouTube Najwa Shihab
"""
data_najwa_shihab.head(8)

video_id                channel_id                  comment_id  \
0  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugx6mhUfDOUhd65fWyh4AaABAg   
1  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgxfMHgX25B2b3jkO4B4AaABAg   
2  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwTpSK1hjkTNQlcT894AaABAg   
3  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgzaWyuIp8PPLQrUOct4AaABAg   
4  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzKC1PcDJZk7TvQ_Qh4AaABAg   
5  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxtzzc3zKJ8nY3Sg7F4AaABAg   
6  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwB34PptWoq1w3z7xN4AaABAg   
7  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugypp7MrvLOhoz9GGUV4AaABAg   

                     author  \
0          @udinlamidun9716   
1            @zakishofi3442   
2            @sandigustiana   
3         @abizardfahry4591   
4           @UmmuAzizah2010   
5            @nursusilo7298   
6  @lilissuryanilimbong7685   
7           @abisopiyan8628   

                                        text_comment  like_count  total_reply  \
0  Siapa yg setuju 3 capres di undang bertiga di ...        3970          332   
1  Saya bukan pendukung siapapun, tapi terimakasi...        3182           91   
2  Timestamp<br><br><a href="https://www.youtube....        2936          124   
3  Siapa yang disini setuju 3 capres dan cawapres...        2531           91   
4  Maa syaa Allaah Pak Anies ❤❤❤❤ <br>Terima kasi...        1542           23   
5  Tak terbantahkan Pak Anis cards yg paling siap...        1501           91   
6  Selaku Ibu Rumah Tangga saya tahu apa yg dibut...        1486          169   
7  Saya dan keluarga berkat debat pertama ini ras...        1452           54   

                                       channel_url      published_date  \
0          http://www.youtube.com/@udinlamidun9716 2024-01-07 17:03:53   
1            http://www.youtube.com/@zakishofi3442 2023-12-12 21:22:08   
2            http://www.youtube.com/@sandigustiana 2024-01-21 21:32:13   
3         http://www.youtube.com/@abizardfahry4591 2024-01-22 00:17:18   
4           http://www.youtube.com/@UmmuAzizah2010 2023-12-12 16:27:52   
5            http://www.youtube.com/@nursusilo7298 2023-12-12 21:00:05   
6  http://www.youtube.com/@lilissuryanilimbong7685 2024-02-04 20:29:41   
7           http://www.youtube.com/@abisopiyan8628 2023-12-12 17:49:04   

         updated_date  
0 2024-01-09 23:18:49  
1 2023-12-12 21:22:08  
2 2024-01-21 21:32:13  
3 2024-01-22 00:17:18  
4 2023-12-12 16:27:52  
5 2023-12-12 21:00:05  
6 2024-02-04 20:29:41  
7 2023-12-12 17:49:04

In [61]:
# Menampilkan jumlah baris data dan kolom yang dimiliki oleh variabel DataFrame "data_najwa_shihab"
data_najwa_shihab.shape

(22999, 10)

Berdasarkan hasil output diatas dapat diketahui jika jumlah baris data yang dimiliki oleh variabel DataFrame "data_najwa_shihab" adalah sebanyak 22.999 data yang terdapat di dalam 10 kolom yang berbeda-beda. Nama-nama sepuluh kolom tersebut yang terdapat di dalam variabel DataFrame "data_najwa_shihab" dapat diketahui menggunakan code script seperti data_najwa_shihab.columns

In [62]:
# Menampilkan nama-nama kolom yang dimiliki oleh variabel DataFrame "data_najwa_shihab"
data_najwa_shihab.columns

Index(['video_id', 'channel_id', 'comment_id', 'author', 'text_comment',
       'like_count', 'total_reply', 'channel_url', 'published_date',
       'updated_date'],
      dtype='object')

Output diatas memberikan kita informasi singkat tentang sepuluh nama-nama kolom yang dimiliki oleh DataFrame "data_najwa_shihab" diantaranya adalah kolom "video_id", "channel_id, "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date", dan kolom terakhir yang dimiliki oleh DataFrame "data_najwa_shihab" adalah kolom "updated_date".

Awalnya sepuluh kolom tersebut merupakan elemen item yang dimiliki oleh resource commentThreads() dan resource comment dari Youtube API v3 yang merupakan salah satu jenis service yang disediakan oleh Google dengan menghasilkan API resmi agar para user dapat mengambil, mengakses dan memanfaatkan data-data publik dari Google seperti pada kasus penelitian ini yang merupakan analisis komentar-komentar yang terdapat pada video debat capres-cawapres dari babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube Najwa Shihab, KOMPASTV dan KPU RI.

Sebagai contoh kolom "video_id" merupakan item dari suatu object yang bernama "snippet" yang dimiliki oleh resource commentThreads(). Dengan mendefinisikan sebuah properti/item "video_id" di dalam resource, maka kita dapat mengambil video id dari seluruh video yang di-upload di media sosial YouTube karena pada dasarnya setiap konten video memiliki nilai unik dan nilai unik tersebut yang merupakan sebuah ID pada video tersebut.

In [63]:
# Menampilkan apakah terdapat data duplikat dari variabel DataFrame "data_najwa_shihab"
data_najwa_shihab.duplicated().sum()

0

Berdasarkan hasil output diatas diketahui jika DataFrame "data_najwa_shihab" tidak memiliki data yang terduplikasi karena sebelumnya terdapat data duplikat dari salah satu video debat capres-cawapres yang ditayangkan oleh channel YouTube Najwa Shihab. Karena jika di dalam DataFrame masih terdapat data duplikat akan menyebabkan kecacatan terhadap data tersebut walaupun hanya memiliki 1 data duplikat di dalam DataFrame.

Selanjutnya saya akan menampilkan apakah di dalam DataFrame "data_najwa_shihab" memiliki missing value/nilai data yang hilang dari masing-masing kolom yang dimiliki oleh variabel DataFrame "data_najwa_shihab".

In [64]:
"""
Menampilkan jumlah missing value/nilai data yang hilang dari setiap
kolom yang ada di dalam variabel DataFrame "data_najwa_shihab"
"""
data_najwa_shihab.isna().sum()

video_id          0
channel_id        0
comment_id        0
author            0
text_comment      0
like_count        0
total_reply       0
channel_url       0
published_date    0
updated_date      0
dtype: int64

In [65]:
"""
Menampilkan beberapa informasi yang dimiliki oleh DataFrame "data_najwa_shihab"
seperti tipe data setiap kolom, jumlah baris data atau kolom yang dimiliki oleh DataFrame,
nama kolom yang dimiliki oleh DataFrame dan apakah masing-masing kolom memiliki missing value/tidak
"""
data_najwa_shihab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22999 entries, 0 to 22998
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        22999 non-null  object        
 1   channel_id      22999 non-null  object        
 2   comment_id      22999 non-null  object        
 3   author          22999 non-null  object        
 4   text_comment    22999 non-null  object        
 5   like_count      22999 non-null  int64         
 6   total_reply     22999 non-null  int64         
 7   channel_url     22999 non-null  object        
 8   published_date  22999 non-null  datetime64[ns]
 9   updated_date    22999 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 1.8+ MB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "data_najwa_shihab" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "data_najwa_shihab" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolom dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden - Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube Najwa Shihab.

2. Variabel DataFrame "data_najwa_shihab" memiliki jumlah data/baris sebanyak 22.999 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang akhirnya disimpan di dalam sebuah file Spreadsheet berformat Microsoft Excel dengan nama file yaitu "Dataset Komentar Video Debat dari YouTube Channel Najwa Shihab.xlsx".

3. DataFrame "data_najwa_shihab" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "data_najwa_shihab" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "data_najwa_shihab" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "data_najwa_shihab" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "data_najwa_shihab" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "data_najwa_shihab" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "data_najwa_shihab" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

### 1.2. Read/Load Dataset from Scraping Result based on Youtube Channel KOMPASTV

In [66]:
"""
Membaca file dataset dari hasil scraping komentar-komentar pada video debat babak
pertama hingga babak kelima capres-cawapres dari channel YouTube KOMPASTV
"""
data_kompastv = pd.read_excel("Dataset/Dataset Komentar Video Debat dari YouTube Channel KOMPASTV.xlsx")


"""
Menampilkan 29 data teratas dari variabel DataFrame "data_kompastv"
yang memiliki jumlah total baris sebanyak 12.500 data yang didapatkan
dari video komentar debat babak pertama hingga babak kelima capres-cawapres di channel YouTube KOMPASTV
"""
data_kompastv.head(29)

video_id                channel_id                  comment_id  \
0   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzTOJaoDFAy69VEmW54AaABAg   
1   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzmTSpOSPp8Aaid8tN4AaABAg   
2   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgytfyaQRBKOJD1MWmJ4AaABAg   
3   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMdksVHA_TUd0b8zZ4AaABAg   
4   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzKN65yuzOuwoV25xR4AaABAg   
5   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwkGCjnkn4qiL2pUq54AaABAg   
6   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwHT6Z026fOcLqsxT54AaABAg   
7   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwEn2xOZrliv0jvyhp4AaABAg   
8   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwpSjBjJZnKzyn5PmN4AaABAg   
9   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugz0wZUiNXc6ccaFu7J4AaABAg   
10  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyGHImkaLkXkDAMqcZ4AaABAg   
11  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyU-X3Lsh3ILboVJ3Z4AaABAg   
12  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwZ5CQXcfQ989V1nWR4AaABAg   
13  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxmaRidkVx6ZNhtXwN4AaABAg   
14  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwWEGrP4EKIKmkiOqN4AaABAg   
15  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxOFelnwKe7HLPP1aN4AaABAg   
16  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyaFu1tTObuurbV-ol4AaABAg   
17  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxaFx3JHpVIapD10S94AaABAg   
18  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugy7In-j-fEWYEmivrJ4AaABAg   
19  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugz44hx8zC79L1OTHWd4AaABAg   
20  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwjqAlc_Ei9wHRT0A94AaABAg   
21  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugxd9sibvHqg_kkjn_N4AaABAg   
22  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMPWdvd-urv1CjC1x4AaABAg   
23  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzkuDUgMUIQRv7Fvot4AaABAg   
24  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugwarfje623O694DOKJ4AaABAg   
25  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzrJP46lxWpF7lsYTd4AaABAg   
26  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyOmbKfPj-MB1FFNIZ4AaABAg   
27  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwqPUGLJwMnvpaNwux4AaABAg   
28  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugzx5I4JHk8B-jKdj_R4AaABAg   

                       author  \
0          @alkhasanisani3619   
1             @MrGaming-fg9qu   
2          @desyanggraeni9380   
3           @aprianisalam7842   
4           @nanananina-eg5vo   
5        @RidhoRamadhan-gm6rp   
6            @user-mf1tc7co8s   
7           @nanasepdiana5531   
8                @FauzanFikli   
9             @titisuchan3357   
10           @PutriPeni-bx3rs   
11       @ahmanalsebayang6366   
12                @nana.edu23   
13      @tatytatydjulaeha2395   
14              @Rissachannel   
15       @JihadSaepuloh-sf9xt   
16           @niosubaktio1995   
17          @hadisuhendro8979   
18  @anisapurialdamayanti7148   
19        @SigitBahtiar-on6gs   
20             @hunterrevew94   
21           @susioktavia2369   
22      @WidhiannisaUlhusna96   
23    @dicky_dentajourney5477   
24          @hambaAllah-ph5gf   
25        @financecompany1582   
26                @prill_syaa   
27       @procrastinating3859   
28           @risnaholisa6882   

                                         text_comment  like_count  \
0   Anies pinter ngomong<br>Ganjar pinter jawab<br...         772   
1   Indonesia tidak kekurangan orang orang pintar&...         607   
2   Disaat Ganjar dan Anies saling menyerang Prabo...         474   
3   Pak Prabowo memang kurang dr hal seperti pak g...         450   
4   Jendral tidak dilatih untuk merangkai kata2 ma...         431   
5   Saya tau rasanya menjadi orang seperti pak pro...         409   
6   Pak prabowo tu keras , disiplin karena mantan ...         338   
7   Pak prabowo tidak pandai merangkai kata - kata...         330   
8   Saya yakin .pak Prabowo sudah tahu keadaan neg...         318   
9   P Prabowo tidak menjatuhkan lawan,,tetap legow...         312   
10  pak prabowo tidak pandai merangkai kata,tapi d...         293 

In [67]:
# Menampilkan jumlah data dan kolom yang dimiliki oleh DataFrame "data_kompastv"
data_kompastv.shape

(12500, 10)

Berdasarkan hasil output diatas dapat diketahui jika jumlah baris data yang dimiliki oleh variabel DataFrame "data_kompastv" adalah sebanyak 12.500 data yang terdapat di dalam 10 kolom yang berbeda-beda. Nama-nama sepuluh kolom tersebut yang terdapat di dalam variabel DataFrame "data_kompastv" dapat diketahui menggunakan code script seperti data_kompastv.columns

In [68]:
# Menampilkan nama-nama kolom yang dimiliki oleh variabel DataFrame "data_kompastv"
data_kompastv.columns

Index(['video_id', 'channel_id', 'comment_id', 'author', 'text_comment',
       'like_count', 'total_reply', 'channel_url', 'published_date',
       'updated_date'],
      dtype='object')

Output diatas memberikan kita informasi singkat tentang sepuluh nama-nama kolom yang dimiliki oleh DataFrame "data_kompastv" diantaranya adalah kolom "video_id", "channel_id, "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date", dan kolom terakhir yang dimiliki oleh DataFrame "data_kompastv" adalah kolom "updated_date".

Awalnya sepuluh kolom tersebut merupakan elemen item yang dimiliki oleh resource commentThreads() dan resource comment dari Youtube API v3 yang merupakan salah satu jenis service yang disediakan oleh Google dengan menghasilkan API resmi agar para user dapat mengambil, mengakses dan memanfaatkan data-data publik yang terdapat diseluruhh service Google seperti pada kasus penelitian ini yang memanfaatkan data-data komentar untuk melakukan analisis sentimen terhadap komentar-komentar yang terdapat pada video debat capres-cawapres dari babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KOMPASTV.

Sebagai contoh kolom "video_id" merupakan item dari suatu object yang bernama "snippet" yang dimiliki oleh resource commentThreads(). Dengan mendefinisikan sebuah properti/item "video_id" di dalam resource, maka kita dapat mengambil video id dari seluruh video yang di-upload di media sosial YouTube karena pada dasarnya setiap konten video memiliki nilai unik dan nilai unik tersebut yang merupakan sebuah ID pada video tersebut.

In [69]:
# Menampilkan apakah terdapat data duplikat dari variabel DataFrame "data_kompastv"
print("Jumlah Data Duplikat dari Variabel DataFrame 'data_kompastv' adalah:", data_kompastv.duplicated().sum())

Jumlah Data Duplikat dari Variabel DataFrame 'data_kompastv' adalah: 0


Berdasarkan hasil output diatas diketahui jika DataFrame "data_kompastv" tidak memiliki data yang terduplikasi bahkan dari sebelumnya juga tidak memiliki data duplikat dimasing-masing DataFrame setelah melakukan proses scraping dari video debat capres - cawapres dari babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KOMPASTV. Karena jika di dalam DataFrame data tersebut memiliki atau masih terdapat data duplikat, maka proses analisis akan bias dan dapatmenyebabkan kecacatan terhadap data tersebut walaupun hanya memiliki 1 data duplikat di dalam DataFrame.

Selanjutnya saya akan menampilkan apakah di dalam DataFrame "data_kompastv" memiliki missing value/nilai data yang hilang dari masing-masing kolom yang dimiliki oleh variabel DataFrame "data_kompastv".

In [70]:
"""
Menampilkan jumlah missing value/nilai data yang hilang dari setiap
kolom yang ada di dalam variabel DataFrame "data_kompastv"
"""
print("Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom pada Variabel DataFrame 'data_kompastv':")
data_kompastv.isna().sum()

Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom pada Variabel DataFrame 'data_kompastv':


video_id          0
channel_id        0
comment_id        0
author            0
text_comment      0
like_count        0
total_reply       0
channel_url       0
published_date    0
updated_date      0
dtype: int64

In [71]:
"""
Menampilkan beberapa informasi yang dimiliki oleh DataFrame "data_kompastv"
seperti tipe data setiap kolom, jumlah baris data atau kolom yang dimiliki oleh DataFrame,
nama kolom yang dimiliki oleh DataFrame dan apakah masing-masing kolom memiliki missing value/tidak
"""
data_kompastv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12500 entries, 0 to 12499
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        12500 non-null  object        
 1   channel_id      12500 non-null  object        
 2   comment_id      12500 non-null  object        
 3   author          12500 non-null  object        
 4   text_comment    12500 non-null  object        
 5   like_count      12500 non-null  int64         
 6   total_reply     12500 non-null  int64         
 7   channel_url     12500 non-null  object        
 8   published_date  12500 non-null  datetime64[ns]
 9   updated_date    12500 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 976.7+ KB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "data_kompastv" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "data_kompastv" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolom dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden - Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KOMPASTV.

2. Variabel DataFrame "data_kompastv" memiliki jumlah data/baris sebanyak 12.500 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang akhirnya disimpan di dalam sebuah file Spreadsheet berformat Microsoft Excel dengan nama file yaitu "Dataset Komentar Video Debat dari YouTube Channel KOMPASTV.xlsx".

3. DataFrame "data_kompastv" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "data_kompastv" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "data_kompastv" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "data_kompastv" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "data_kompastv" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "data_kompastv" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "data_kompastv" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

In [72]:
# Menampilkan 7 data terakhir dari 12.500 data yang dimiliki oleh DataFrame "data_kompastv"
data_kompastv.tail(7)

video_id                channel_id                  comment_id  \
12493  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwPBUzZUg8rQ8yalUx4AaABAg   
12494  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgztuSxn-AH6w8TZ0oN4AaABAg   
12495  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugx_okVRRVsLfNPVYE14AaABAg   
12496  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyPayZtl07X4nm426B4AaABAg   
12497  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugzllk4WmVZlviVlA-54AaABAg   
12498  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwwfOip87BLyJjCGSp4AaABAg   
12499  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwEDftvvE9Yu1x-hDd4AaABAg   

                           author  \
12493  @ND_All_Tutorial_Channel27   
12494            @misbahhudin2606   
12495            @user-dz8vp2ge4b   
12496            @devisusanti6630   
12497           @dimassetiadi1152   
12498            @user-pq7np6ie4o   
12499                @fahimhd2141   

                                            text_comment  like_count  \
12493      1 pinter omong<br>2 tegas<br>3 petugas partai           0   
12494                          Riweh oy anis yg enak mah           0   
12495  Visi misi belau dari awal masalah HAM belau te...           0   
12496  Itu pendukung Prabowo liar banget sih berisik ...           0   
12497                      Indonesia butuh bapak Prabowo           0   
12498                                Ganjar 3 yes menang           0   
12499  banyak yg bilang anies banyak omong, yap betul...           0   

       total_reply                                        channel_url  \
12493            0  http://www.youtube.com/@ND_All_Tutorial_Channel27   
12494            0            http://www.youtube.com/@misbahhudin2606   
12495            0            http://www.youtube.com/@user-dz8vp2ge4b   
12496            0            http://www.youtube.com/@devisusanti6630   
12497            0           http://www.youtube.com/@dimassetiadi1152   
12498            0            http://www.youtube.com/@user-pq7np6ie4o   
12499            0                http://www.youtube.com/@fahimhd2141   

           published_date        updated_date  
12493 2023-12-14 20:20:25 2023-12-14 20:20:25  
12494 2023-12-14 20:22:19 2023-12-14 20:22:19  
12495 2023-12-14 20:26:44 2023-12-14 20:26:44  
12496 2023-12-14 20:30:09 2023-12-14 20:30:09  
12497 2023-12-14 20:45:01 2023-12-14 20:45:01  
12498 2023-12-14 20:46:04 2023-12-14 20:46:04  
12499 2024-02-10 07:59:39 2024-02-10 07:59:39

### 1.3. Read/Load Dataset from Scraping Result based on Youtube Channel KPU RI

In [73]:
"""
Membaca file dataset dari hasil scraping komentar-komentar pada video debat babak
pertama hingga babak kelima capres-cawapres dari channel YouTube KPU RI
"""
data_kpu_ri = pd.read_excel("Dataset/Dataset Komentar Video Debat dari YouTube Channel KPU RI.xlsx")


"""
Menampilkan 16 data terbawah dari variabel DataFrame "data_kpu_ri"
yang memiliki jumlah total baris sebanyak 12.740 data yang didapatkan
dari video komentar debat babak pertama hingga babak kelima capres-cawapres di channel YouTube KPU RI
"""
data_kpu_ri.tail(16)

video_id                channel_id                  comment_id  \
12724  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz9rbD9wBauIXmKM3Z4AaABAg   
12725  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxe84yZ_9z8SAYAlL94AaABAg   
12726  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxBqSGFH9aq_4X7OMp4AaABAg   
12727  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz87rZnNhvxza_H7ox4AaABAg   
12728  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwKEOvb11y7Q0pRrwJ4AaABAg   
12729  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzNI662P7I7_s3754Z4AaABAg   
12730  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxvcd8EzUqzfAqTHE54AaABAg   
12731  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgypMisPlgKHZ52LwQl4AaABAg   
12732  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxjoDbjrcwuPl9ckDp4AaABAg   
12733  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyBA7lugeuSwOzwqk54AaABAg   
12734  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwWRqxy1PFBhjXJ7z14AaABAg   
12735  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy-jylcfVdKtTBnbAp4AaABAg   
12736  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxRNOjXy2il2JGsZJF4AaABAg   
12737  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxOwgAbtyVG412f0314AaABAg   
12738  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzJFZj2KoWhCT_MooF4AaABAg   
12739  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx4nJkrQUve_UheMdp4AaABAg   

                          author  \
12724          @SitiNurhasanah69   
12725          @luqmanelghoni266   
12726          @jobintbintang386   
12727               @NyimasAyu43   
12728           @Rani_difarani98   
12729  @dinarikaanggidesylia1408   
12730          @AgungWidianto216   
12731                @Rani_ani32   
12732            @KarangPilang-1   
12733         @tauvikhidayat9031   
12734                  @Krtpl119   
12735         @risyanasrilah4295   
12736            @gikamojang2023   
12737            @SpillProducInd   
12738                 @elfli4250   
12739               @opanboc5038   

                                            text_comment  like_count  \
12724    GASPOL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12725   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12726  Semakin tdk yakin kpd PRAGIB setelah 2x nyalon...           0   
12727     GASPOL BRO!PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12728     GASPOL BRO!PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12729  Semoga pasangan AMIN menjadi pemimpin yang mam...           0   
12730   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12731     GASPOL BRO!PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12732   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12733  Sosok pemimpin seperti pak anies lah yang di b...           0   
12734   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12735    GASPOL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
12736  yakin dengan program kerja PRAGIB...<br>lanjut...           0   
12737  Program kerja yg d lakukan oleh pak prabowo da...           0   
12738                                 Gaspol paslon no 2           0   
12739  Gak ada Keraguan Untuk Pasangan AMIN....Seanda...           0   

       total_reply                                       channel_url  \
12724            0          http://www.youtube.com/@SitiNurhasanah69   
12725            0          http://www.youtube.com/@luqmanelghoni266   
12726            0          http://www.youtube.com/@jobintbintang386   
12727            0               http://www.youtube.com/@NyimasAyu43   
12728            0           http://www.youtube.com/@Rani_difarani98   
12729            0  http://www.youtube.com/@dinarikaanggidesylia1408   
12730            0          http://www.youtube.com/@AgungWidianto216   
12731            0                http://www.youtube.com/@Rani_ani32   
12732            0            http://www.youtube.com/@KarangPilang-1   
12733            0         http://www.youtube.com/@tauvikhidayat9031   
12734            0                  http://www.youtube.com/@Krtpl119   
12735          

In [74]:
# Menampilkan jumlah data dan kolom yang dimiliki oleh DataFrame "data_kpu_ri"
data_kpu_ri.shape

(12740, 10)

Berdasarkan hasil output diatas dapat diketahui jika jumlah baris data yang dimiliki oleh variabel DataFrame "data_kpu_ri" adalah sebanyak 12.740 data yang terdapat di dalam 10 kolom yang berbeda-beda. Nama-nama sepuluh kolom tersebut yang terdapat di dalam variabel DataFrame "data_kpu_ri" dapat diketahui menggunakan code script seperti data_kpu_ri.columns

In [75]:
# Menampilkan nama-nama kolom yang dimiliki oleh variabel DataFrame "data_kpu_ri"
data_kpu_ri.columns

Index(['video_id', 'channel_id', 'comment_id', 'author', 'text_comment',
       'like_count', 'total_reply', 'channel_url', 'published_date',
       'updated_date'],
      dtype='object')

Output diatas memberikan kita informasi singkat tentang sepuluh nama-nama kolom yang dimiliki oleh DataFrame "data_kpu_ri" diantaranya adalah kolom "video_id", "channel_id, "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date", dan kolom terakhir yang dimiliki oleh DataFrame "data_kompastv" adalah kolom "updated_date".

Awalnya sepuluh kolom tersebut merupakan elemen item yang dimiliki oleh resource commentThreads() dan resource comment dari Youtube API v3 yang merupakan salah satu jenis service yang disediakan oleh Google dengan menghasilkan API resmi agar para user dapat mengambil, mengakses dan memanfaatkan data-data publik yang terdapat diseluruhh service Google seperti pada kasus penelitian ini yang memanfaatkan data-data komentar untuk melakukan analisis sentimen terhadap komentar-komentar yang terdapat pada video debat capres-cawapres dari babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KPU RI.

Sebagai contoh kolom "video_id" merupakan item dari suatu object yang bernama "snippet" yang dimiliki oleh resource commentThreads(). Dengan mendefinisikan sebuah properti/item "video_id" di dalam resource, maka kita dapat mengambil video id dari seluruh video yang di-upload di media sosial YouTube karena pada dasarnya setiap konten video memiliki nilai unik dan nilai unik tersebut yang merupakan sebuah ID pada video tersebut.

In [76]:
# Menampilkan apakah terdapat data duplikat dari variabel DataFrame "data_kpu_ri"
print("Jumlah Data Duplikat dari Variabel DataFrame 'data_kpu_ri' adalah:", data_kpu_ri.duplicated().sum())

Jumlah Data Duplikat dari Variabel DataFrame 'data_kpu_ri' adalah: 0


Berdasarkan hasil output diatas diketahui jika DataFrame "data_kpu_ri" tidak memiliki data yang terduplikasi bahkan dari sebelumnya juga tidak memiliki data duplikat dimasing-masing DataFrame setelah melakukan proses scraping dari video debat capres - cawapres dari babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KPU RI. Karena jika di dalam DataFrame data tersebut memiliki atau masih terdapat data duplikat, maka proses analisis akan bias dan dapatmenyebabkan kecacatan terhadap data tersebut walaupun hanya memiliki 1 data duplikat di dalam DataFrame.

Selanjutnya saya akan menampilkan apakah di dalam DataFrame "data_kpu_ri" memiliki missing value/nilai data yang hilang dari masing-masing kolom yang dimiliki oleh variabel DataFrame "data_kpu_ri".

In [77]:
"""
Menampilkan jumlah missing value/nilai data yang hilang dari setiap
kolom yang ada di dalam variabel DataFrame "data_kpu_ri"
"""
print("Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom pada Variabel DataFrame 'data_kpu_ri':")
data_kpu_ri.isna().sum()

Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom pada Variabel DataFrame 'data_kpu_ri':


video_id           0
channel_id         0
comment_id         0
author            10
text_comment       0
like_count         0
total_reply        0
channel_url       10
published_date     0
updated_date       0
dtype: int64

Berdasarkan hasil diatas diketahui jika ternyata variabel DataFrame "data_kpu_ri" memiliki missing value di dalam kolom author dan channel_url. Jumlah baris/data yang terindikasi memiliki missing value didalamnya adalah 10 baris data saja. Tetapi walaupun begitu kehilangan data/informasi sebanyak 10 data tetap akan mempengaruhi analisis serta Model Machine Learning yang akan diimplementasikan terlebih penelitian kali ini bersifat sentimen analisis yang berarti akan melakukan proses klasifikasi berdasarkan sentimen terhadap komentar dari kumpulan video debat capres-cawapres tahun 2024.

Terdapat beberapa cara dalam menangani data yang hilang di dalam data science dan cara paling termudah adalah dengan menghilangkan data yang hilang tersebut dari daftar baris di DataFrame, baik data numerik dengan tipe data Integer atau tipe data Float. Cara kedua adalah melakukan proses imputation (imputasi) atau mengganti missing value tersebut dengan nilai lainnya. Sebagai contoh jika suatu kolom yang memiliki data numerik dengan tipe data Integer terdapat beberapa missing value, maka dengan melakukan pross imputation missing value tersebut akan digantikan dengan nilai statistik seperti mean, mode, max, min atau nilai numerik dari baris data sebelumnya/sesudahnya. Atau cara ketiga dalam menangani missing value disuatu kolom adalah dapat menghapus kolom tersebut namun cara ini sangat tidak efektif dan tidak dianjurkan kecuali jika data yang hilang di dalam kolom tersebut sangatlah banyak.

Maka dari itu, cara yang akan digunakan di dalam penelitian kali ini untuk menangani missing value yang ada di dalam kolom "author" dan kolom "channel_url" pada variabel DataFrame "data_kpu_ri" adalah cara kedua karena saya tidak menginginkan untuk menghapus semua informasi, baik menghapus data per baris atau menghapus kolom yang memiliki missing value dan dalam hal ini kolom "author" dan "channel_url" akan dihilangkan dari variabel DataFrame "data_kpu_ri". Karena saya akan melakukan proses imputation untuk dua kolom yang bertipe data String, maka penelitian ini akan menggunakan proses imputation dengan cara mengubah setiap baris data yang memiliki missing value dengan nilai kategorik (nilai string) yang spesifik yaitu Unknown atau tidak diketahui sebagai representasi dari data-data yang hilang. Sehingga data-data yang sebelumnya tertulis sebagai NaN atau Not a Number (Missing Value), baik data NaN yang ada di dalam kolom "author" dan "channel_url" akan berubah menjadi nilai data Unknown.

In [78]:
# Menampilkan 10 data yang hilang dari variabel DataFrame "data_kpu_ri", baik dari kolom "author" maupun kolom "channel_url"

"""
Memfilter data dari variabel DataFrame "data_kpu_ri" untuk mengambil
missing value yang dimiliki oleh kolom "author" dan "channel_url"
dan disimpan di dalam variabel baru yang bernama "data_kpu_ri_null"
"""
data_kpu_ri_null = data_kpu_ri[data_kpu_ri["author"].isna()]

data_kpu_ri_null

video_id                channel_id                  comment_id  \
1035  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzKGFgTPC-e_oSNCSN4AaABAg   
1683  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwHFarW3U_GX7-1pqR4AaABAg   
1906  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxc-4ccVymFdQ9G2bx4AaABAg   
3645  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugzy5-LSm1AO2C1DBF14AaABAg   
3649  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxMajpcMqHLHEC4pLd4AaABAg   
3651  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw0POcCCtHsU6ks0lB4AaABAg   
6092  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz0lQWuf1ASK5iT-Qh4AaABAg   
6374  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyTZaOkJydE2pmXz_N4AaABAg   
6398  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzmTYLWbXPg4G02K-54AaABAg   
6402  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyP73VqutK3bfej-VN4AaABAg   

     author                                       text_comment  like_count  \
1035    NaN  Pindah dri 1 ke 2....jadilah pahlawan dan peme...          11   
1683    NaN  Cuman 1 slogan ,ingin perubahan (keluar dari r...           5   
1906    NaN  intinya kita lihat kalau sudah menjabat...kala...           4   
3645    NaN  Tidak memberikan hasil untuk menimbang / mamil...           1   
3649    NaN  Mantapp... Untuk Indonesia lbh baik jgn lupa g...           1   
3651    NaN  Semakin yakin milih PRABOWO jadi PRESIDEN INDO...           1   
6092    NaN                          Retorika kata kata saja 😂           0   
6374    NaN              Berikutnya gak perlu ada suporter,min           0   
6398    NaN                                 Aniiis.. 1 putaran           0   
6402    NaN                      Prabowo presiden 2024🩵🩵✌️✌️✌️           0   

      total_reply channel_url      published_date        updated_date  
1035            2         NaN 2024-01-07 14:54:43 2024-01-07 14:54:43  
1683            0         NaN 2024-01-07 14:55:05 2024-01-07 14:55:05  
1906            0         NaN 2024-01-07 14:55:18 2024-01-07 14:55:18  
3645            0         NaN 2024-01-07 14:53:54 2024-01-07 14:53:54  
3649            0         NaN 2024-01-07 14:54:16 2024-01-07 14:54:16  
3651            0         NaN 2024-01-07 14:54:41 2024-01-07 14:54:41  
6092            0         NaN 2024-01-07 14:54:04 2024-01-07 14:54:04  
6374            0         NaN 2024-01-07 14:54:13 2024-01-07 14:54:13  
6398            0         NaN 2024-01-07 14:55:17 2024-01-07 14:55:17  
6402            0         NaN 2024-01-07 14:55:26 2024-01-07 14:55:26

Output diatas merupakan 10 data baik dari kolom "author" dan "channel_url" yang memiliki missing value di dalam 2 kolom tersebut. Indeks baris data ke-1035, 1683, 1906, 3645, 3649, 3651, 6092, 6374, 6398 dan indeks data terakhir yang memiliki missing value didalamnya adalah indeks ke-6402. Kesepuluh indeks baris data tersebut memiliki nilai NaN atau Not a Number, sebuah nilai yang merepresentasikan bahwa data tersebut tidak memiliki nilai alias missing value.

Seperti yang sudah diinformasikan sebelumnya bahwasanya dalam penelitian kali ini, saya akan menggantikan nilai-nilai NaN yang dimiliki oleh kolom "author" dan "channel_url" menjadi nilai kategorik "Unknown" supaya data-data yang terkandung di kolom lainnya tidak terbuang sehingga dapat mempertahankan keaslian informasi dan nantinya akan sangat berguna untuk Algoritma Machine Learning dalam mempelajari pola-pola serta juga akan sangat berguna bagi tahapan text processing seperti tokenization, stopwords, dll., karena akan mempelajari banyak pola teks Bahasa Indonesia.

In [81]:
"""
Mentransformasikan nilai-nilai data yang hilang (missing value)
pada kolom "author" dan "channel_url" yang dimiliki variabel DataFrame "data_kpu_ri"
dengan sebuah nilai string/kategorik yang ditentukan (spesific string value) yaitu Unknown
untuk setiap data yang sebelumnya memiliki nilai NaN pada baris data tersebut
"""

def change_missing_value_data(data_kpu_ri):
    """
    Mengubah missing value di dalam kolom "author" menjadi spesifik string yakni "Unknown"
    sehingga dapat diasumsikan jika nama user/akun YouTube-nya tidak diketahui
    """
    data_kpu_ri["author"] = data_kpu_ri["author"].fillna("Unknown")
    
    """
    Mengubah missing value di dalam kolom "channel_url" juga menjadi spesifik string yakni "Unknown"
    sehingga dapat diasumsikan jika url/link yang mengarahkan ke profile user/akun tidak diketahui
    """
    data_kpu_ri["channel_url"] = data_kpu_ri["channel_url"].fillna("Unknown")
    
    
    return change_missing_value_data

change_missing_value_data(data_kpu_ri)

# Menampilkan kembali jumlah nilai data yang hilang dari masing-masing kolom pada variabel DataFrame "data_kpu_ri"
print("Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom setelah Missing Value Dihilangkan adalah:")
data_kpu_ri.isnull().sum()

Jumlah Nilai Data yang Hilang (Missing Value) dari Masing-Masing Kolom setelah Missing Value Dihilangkan adalah:


video_id          0
channel_id        0
comment_id        0
author            0
text_comment      0
like_count        0
total_reply       0
channel_url       0
published_date    0
updated_date      0
dtype: int64

Berdasarkan output diatas diketahui bahwasanya masing-masing kolom khususnya kolom "author" dan kolom "channel_url" sudah tidak memiliki nilai NaN yang menjadi representasi bahwa pada data tersebut memiliki nilai data yang hilang (missing value).

In [83]:
# Menampilkan 13 data teratas yang dimiliki oleh variabel DataFrame "data_kpu_ri"
data_kpu_ri.head(13)

video_id                channel_id                  comment_id  \
0   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy7bfVbuDxBgae9eVp4AaABAg   
1   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz8GZ3Lfw7X9kYnqSV4AaABAg   
2   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyBL7L6U5bXWuuzlap4AaABAg   
3   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzXkcSL0MuTJL6ZpY54AaABAg   
4   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwXBn-0QCB6GntoHY94AaABAg   
5   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzF-687IdOdv5qeEsp4AaABAg   
6   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxbXCPeUDV_zPUhZjF4AaABAg   
7   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw-FFWX0IY1b5sW77x4AaABAg   
8   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw_i1DjvR-QcFZOWFR4AaABAg   
9   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyLvJxPNZznf4VD_hF4AaABAg   
10  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwOYSRHpyihNxbUL_l4AaABAg   
11  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzE0Z69Qxhmo7W7rG54AaABAg   
12  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy8BsMRwY9YzhEKdUN4AaABAg   

                   author                                       text_comment  \
0        @nabilbudiman271  <a href="https://www.youtube.com/watch?v=KJdt-...   
1      @anisluthfiani9056  Abah terimakasih sudah mewarnai demokrasi ini ...   
2        @nabilbudiman271  <a href="https://www.youtube.com/watch?v=YzC82...   
3        @faridafasya9815  Kami rindu presiden seperti pak Anies semoga A...   
4          @Teman_Bambank  Mari kita lihat sampai akhir, apakah KPU benar...   
5          @rizkiindah479  Dulu dukung Prabowo Sandi, pasangan keren dan ...   
6        @nabilbudiman271  <a href="https://www.youtube.com/watch?v=yNO0Y...   
7            @waryati5554  Saya sebagai org awam pun sangat mengerti kena...   
8     @ajipratamaputra815  Ingat kata Rocky Gerung<br>Argumen harus di ja...   
9   @angkadandata_channel  Smg presiden yg terpilih nantinya bisa memimpi...   
10        @sodrishandy545  Semoga Allah hadirkan pemimpin yang adil bagi ...   
11           @robinong153  Terus terang..debat hari ini lebih berkualitas...   
12           @kajikampung  Saya sangat suka dengan para pengkritik yg int...   

    like_count  total_reply                                   channel_url  \
0         1590          122       http://www.youtube.com/@nabilbudiman271   
1          885          227     http://www.youtube.com/@anisluthfiani9056   
2          750           73       http://www.youtube.com/@nabilbudiman271   
3          670           33       http://www.youtube.com/@faridafasya9815   
4          631           66         http://www.youtube.com/@Teman_Bambank   
5          568           37         http://www.youtube.com/@rizkiindah479   
6          546           26       http://www.youtube.com/@nabilbudiman271   
7          432           90           http://www.youtube.com/@waryati5554   
8          403           34    http://www.youtube.com/@ajipratamaputra815   
9          384           47  http://www.youtube.com/@angkadandata_channel   
10         376           10        http://www.youtube.com/@sodrishandy545   
11         326           20           http://www.youtube.com/@robinong153   
12         294           19           http://www.youtube.com/@kajikampung   

        published_date        updated_date  
0  2024-01-07 15:37:05 2024-01-07 15:37:05  
1  2024-02-04 16:12:53 2024-02-04 16:12:53  
2  2023-12-22 15:35:34 2023-12-23 02:18:45  
3  2023-12-12 15:24:47 2023-12-12 15:24:47  
4  2024-01-07 15:20:09 2024-01-07 15:20:09  
5  2023-12-12 15:47:19 2023-12-12 15:47:19  
6  2023-12-12 20:40:59 2023-12-13 02:44:18  
7  2024-01-08 03:40:07 2024-01-08 03:40:07  
8  2023-12-12 16:02:30 2023-12-12 16:02:30  
9  2024-01-08 08:12:00 2024-01-08 08:12:00  
10 2024-02-04 15:27:34 2024-02-04 15:27:34  
11 2024-02-04 16:34:43 2024-02-04 16:34:43  
12 2024-01-21 16:42:36 2024-01-21 16:42:36

In [85]:
"""
Menampilkan beberapa informasi basic yang terdapat di dalam variabel DataFrame "data_kpu_ri"
seperti tipe data yang dimiliki oleh masing-masing kolom, jumlah baris data dan kolom yang dimiliki oleh DataFrame,
terdapatnya missing value di setiap kolom/tidak serta nama-nama kolom yang terdapat di dalam DataFrame "data_kpu_ri"
"""
data_kpu_ri.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12740 entries, 0 to 12739
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        12740 non-null  object        
 1   channel_id      12740 non-null  object        
 2   comment_id      12740 non-null  object        
 3   author          12740 non-null  object        
 4   text_comment    12740 non-null  object        
 5   like_count      12740 non-null  int64         
 6   total_reply     12740 non-null  int64         
 7   channel_url     12740 non-null  object        
 8   published_date  12740 non-null  datetime64[ns]
 9   updated_date    12740 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 995.4+ KB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "data_kpu_ri" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "data_kpu_ri" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolom dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden - Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KPU RI.

2. Variabel DataFrame "data_kpu_ri" memiliki jumlah data/baris sebanyak 12.740 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang akhirnya disimpan di dalam sebuah file Spreadsheet berformat Microsoft Excel dengan nama file yaitu "Dataset Komentar Video Debat dari YouTube Channel KPU RI.xlsx".

3. DataFrame "data_kpu_ri" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "data_kpu_ri" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "data_kpu_ri" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "data_kpu_ri" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "data_kpu_ri" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "data_kpu_ri" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "data_kompastv" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

### 1.4. Combine Data from Three DataFrame Variables into 1 Centralized DataFrame and Save/Export to Spreadsheet File

In [86]:
comments_debate_df = pd.concat([data_najwa_shihab, data_kompastv, data_kpu_ri], ignore_index = True)

"""
Menampilkan 19 data teratas yang dimiliki oleh variabel DataFrame hasil
gabungan dari variabel DataFrame "data_najwa_shihab", "data_kompastv" dan "data_kpu_ri"
yang bernama "comments_debate_df"
"""
comments_debate_df.head(19)

video_id                channel_id                  comment_id  \
0   Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugx6mhUfDOUhd65fWyh4AaABAg   
1   gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgxfMHgX25B2b3jkO4B4AaABAg   
2   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwTpSK1hjkTNQlcT894AaABAg   
3   tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgzaWyuIp8PPLQrUOct4AaABAg   
4   gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzKC1PcDJZk7TvQ_Qh4AaABAg   
5   gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugxtzzc3zKJ8nY3Sg7F4AaABAg   
6   JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgwB34PptWoq1w3z7xN4AaABAg   
7   gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  Ugypp7MrvLOhoz9GGUV4AaABAg   
8   Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugw1wHlLE405w38TGhx4AaABAg   
9   p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgyJX6diRTVoG-ulwqB4AaABAg   
10  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  Ugwru57KZMbtoy1Y-Wp4AaABAg   
11  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  Ugxbn5S7DiuMkSkSbB94AaABAg   
12  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgyzdGCruWPoVqZjMul4AaABAg   
13  tEbRB5xxOPI  UCo8h2TY_uBkAVUIc14m_KCA  UgwRwfdnlejN5_d0iCp4AaABAg   
14  p-vbZkT3Yck  UCo8h2TY_uBkAVUIc14m_KCA  UgzO_HcRVJSQ7ZrHCel4AaABAg   
15  Sbjsg3AFx00  UCo8h2TY_uBkAVUIc14m_KCA  UgxrCbnzIEfsCua9aLZ4AaABAg   
16  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgxcIoHuK_A7ifquMpJ4AaABAg   
17  gUz_MgdwKg0  UCo8h2TY_uBkAVUIc14m_KCA  UgzwCUAh7kTbyV3l5TJ4AaABAg   
18  JFl2fOTBNIE  UCo8h2TY_uBkAVUIc14m_KCA  UgzrVJ1HRPmYF26uAOx4AaABAg   

                      author  \
0           @udinlamidun9716   
1             @zakishofi3442   
2             @sandigustiana   
3          @abizardfahry4591   
4            @UmmuAzizah2010   
5             @nursusilo7298   
6   @lilissuryanilimbong7685   
7            @abisopiyan8628   
8          @ameliaachtar1904   
9             @luluk_layalie   
10  @MuhammadDermawanSusanto   
11         @erickvalvolt3622   
12       @chickennoodles1740   
13           @RickyFahrezi13   
14         @ichadonnelly4634   
15          @idincelebez3554   
16          @Millenial_IQ181   
17         @rizkyardianto984   
18           @edysupaidi8586   

                                         text_comment  like_count  \
0   Siapa yg setuju 3 capres di undang bertiga di ...        3970   
1   Saya bukan pendukung siapapun, tapi terimakasi...        3182   
2   Timestamp<br><br><a href="https://www.youtube....        2936   
3   Siapa yang disini setuju 3 capres dan cawapres...        2531   
4   Maa syaa Allaah Pak Anies ❤❤❤❤ <br>Terima kasi...        1542   
5   Tak terbantahkan Pak Anis cards yg paling siap...        1501   
6   Selaku Ibu Rumah Tangga saya tahu apa yg dibut...        1486   
7   Saya dan keluarga berkat debat pertama ini ras...        1452   
8   Mudah2an pemilunya jujur, jadi sesuai pilihan ...        1396   
9   Jujur, gw terpukau banget sama Mas Gibran. Ben...        1253   
10  Pak Anies sadar bahwa Pak Prabowo secara teori...        1203   
11  awalnya ragu milih 02 krna wakilnya. akhirnya ...        1133   
12  🇲🇨 <a href="https://www.youtube.com/watch?v=gU...        1130   
13  Saya pendukung AMIN.<br>Saya salut kpd PAK mah...        1110   
14  Terbukti yang muda belum tentu tidak berkualit...         988   
15  Sangat setujuh sekali kalau mereka debat terbu...         984   
16  Terimakasih Bapak H. Anies Rasyid Baswedan, S....         939   
17  Pujian untuk Anies Baswedan tidak hanya pada k...         846   
18  Aplous buat Paslon 01, sangat bijaksana dalam ...         838   

    total_reply                                      channel_url  \
0           332          http://www.youtube.com/@udinlamidun9716   
1            91            http://www.youtube.com/@zakishofi3442   
2           124            http://www.youtube.com/@sandigustiana   
3            91         http://www.youtube.com/@abizardfahry4591   
4            23           http://www.youtube.com/@UmmuAzizah2010   
5            91            http://www.youtube.com/@nursusilo7298   
6           169  http://www.youtube.com/@lilissu

In [89]:
# Menampilkan 11 data terbawah yang dimiliki oleh variabel DataFrame "comments_debate_df"
comments_debate_df.tail(11)

video_id                channel_id                  comment_id  \
48228  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzNI662P7I7_s3754Z4AaABAg   
48229  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxvcd8EzUqzfAqTHE54AaABAg   
48230  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgypMisPlgKHZ52LwQl4AaABAg   
48231  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxjoDbjrcwuPl9ckDp4AaABAg   
48232  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyBA7lugeuSwOzwqk54AaABAg   
48233  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwWRqxy1PFBhjXJ7z14AaABAg   
48234  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy-jylcfVdKtTBnbAp4AaABAg   
48235  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxRNOjXy2il2JGsZJF4AaABAg   
48236  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxOwgAbtyVG412f0314AaABAg   
48237  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzJFZj2KoWhCT_MooF4AaABAg   
48238  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx4nJkrQUve_UheMdp4AaABAg   

                          author  \
48228  @dinarikaanggidesylia1408   
48229          @AgungWidianto216   
48230                @Rani_ani32   
48231            @KarangPilang-1   
48232         @tauvikhidayat9031   
48233                  @Krtpl119   
48234         @risyanasrilah4295   
48235            @gikamojang2023   
48236            @SpillProducInd   
48237                 @elfli4250   
48238               @opanboc5038   

                                            text_comment  like_count  \
48228  Semoga pasangan AMIN menjadi pemimpin yang mam...           0   
48229   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
48230     GASPOL BRO!PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
48231   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
48232  Sosok pemimpin seperti pak anies lah yang di b...           0   
48233   GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
48234    GASPOL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU           0   
48235  yakin dengan program kerja PRAGIB...<br>lanjut...           0   
48236  Program kerja yg d lakukan oleh pak prabowo da...           0   
48237                                 Gaspol paslon no 2           0   
48238  Gak ada Keraguan Untuk Pasangan AMIN....Seanda...           0   

       total_reply                                       channel_url  \
48228            0  http://www.youtube.com/@dinarikaanggidesylia1408   
48229            0          http://www.youtube.com/@AgungWidianto216   
48230            0                http://www.youtube.com/@Rani_ani32   
48231            0            http://www.youtube.com/@KarangPilang-1   
48232            0         http://www.youtube.com/@tauvikhidayat9031   
48233            0                  http://www.youtube.com/@Krtpl119   
48234            0         http://www.youtube.com/@risyanasrilah4295   
48235            0            http://www.youtube.com/@gikamojang2023   
48236            0            http://www.youtube.com/@SpillProducInd   
48237            0                 http://www.youtube.com/@elfli4250   
48238            0               http://www.youtube.com/@opanboc5038   

           published_date        updated_date  
48228 2023-12-13 12:35:05 2023-12-13 12:35:05  
48229 2023-12-13 12:35:05 2023-12-13 12:35:05  
48230 2023-12-13 12:35:34 2023-12-13 12:35:34  
48231 2023-12-13 12:36:05 2023-12-13 12:36:05  
48232 2023-12-13 12:36:33 2023-12-13 12:36:33  
48233 2023-12-13 12:37:07 2023-12-13 12:37:07  
48234 2023-12-13 12:37:31 2023-12-13 12:37:31  
48235 2023-12-13 12:37:32 2023-12-13 12:37:32  
48236 2023-12-13 12:38:48 2023-12-13 12:38:48  
48237 2023-12-13 12:53:09 2023-12-13 12:53:09  
48238 2024-02-04 18:22:22 2024-02-04 18:22:22

Berdasarkan code diatas, saya telah berhasil menggabungkan 3 variabel DataFrame yang masing-masing DataFrame memiliki data/informasi komentarnya masing-masing. Sebagai contoh pada variabel DataFrame "data_najwa_shihab" memiliki 22.999 data/informasi komentar yang berhasil diambil dari video debat babak pertama hingga babak kelima capres - cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel Najwa Shihab. Selanjutnya adalah variabel DataFrame "data_kompastv" yang memiliki jumlah data sebanyak 12.500 data yang bersumber dari komentar-komentar pada video debat babak pertama hingga kelima capres - cawapres Republik Indonesia tahun 2024 yang ditayangkan dan di-upload oleh channel YouTube KOMPASTV. Terakhir adalah variabel DataFrame "data_kpu_ri" memiliki jumlah data sebanyak 12.740 data yang diambil dari video debat calon Presiden - Wakil Presiden Republik Indonesia tahun 2024 dari babak pertama debat hingga babak kelima yang ditayangkan oleh channel KPU RI.

Untuk mengecek dan memastikan apakah penggabungan 3variabel DataFrame berhasil dilakukan, saya telah menampilkan 11 data terbawah yang dimiliki oleh DataFrame "comments_debate_df" karena jika proses penggabungan ketiga variabel DataFrame menjadi 1 DataFrame yang tersentralisasi berhasil dilakukan maka 11 data terbawah seharusnya berasal dari data-data komentar yang diambil (di-scraping) dari variabel DataFrame "data_kpu_ri" yang sumber-sumber datanya diambil dari video debat babak pertama hingga babak kelima Calon Presiden - Wakl Presiden Republik Indonesia tahun 2024 yang di-upload oleh YouTube Channel KPU RI.

Sebagai penjelasan tambahan saat saya melakukan proses penggabungan ketiga variabel DataFrame menjadi sebuah DataFrame yang tersentralisasi dengan nama DataFrame yaitu "comments_debate_df", saya menggunakan sebuah parameter di dalam fungsi pandas pd.concat() yaitu parameter yang bernama ignore_index yang bernilai True (sebuah nilai dari tipe data Boolean). Parameter tersebut saya gunakan (ignore_index = True) bertujuan agar indeks pada DataFrame "comments_debate_df" tidak terduplikasi dan tidak mengulangi nilai indeks yang sama dari indeks awal.

Sebagai contoh, variabel DataFrame "data_kompastv" memiliki sebuah nilai awal indeks yaitu 0 (indeks data 0 mengartikan jika data tersebut merupakan baris pertama yang ada di dalam data tersebut) dan indeks terakhir pada variabel DataFrame "data_kompastv" adalah indeks ke-12499 yang mengartikan bahwa baris data tersebut merupakan baris data ke-12.500. Lalu karena DataFrame "data_kompastv" digabungkan dengan DataFrame "data_kpu_ri", maka baris data selanjutnya sudah pasti merupakan data yang dimiliki oleh indeks data ke-0/baris pertama dari DataFrame "data_kpu_ri". Jika tidak menggunakan parameter ignore_index yang bernilai True, maka variabel DataFrame "comments_debate_df" yang telah menjadi DataFrame tersentralisasi tidak melanjutkan nilai indeks yang selanjutnya yaitu indeks data ke-12500 yang merupakan data pertama/indeks data ke-0 dari variabel DataFrame "data_kpu_ri".

Jadi secara tidak langsung akan ada 2 baris data/2 indeks yang memiliki nilai indeks ke-0 jika DataFrame "comments_debate_df" berisikan data-data komentar dari variabel DataFrame "data_kompastv" dan variabel DataFrame "data_kpu_ri". Intinya adalah jika DataFrame yang tersentralisasi ini (DataFrame "comments_debate_df") berisikan data-data komentar hasil penggabungan ketiga variabel DataFrame yang berbeda, maka jika tidak menggunakan parameter ignore_index bernilai True, maka DataFrame "comments_debate_df" akan memiliki 3 indeks data ke-0, termasuk variabel DataFrame "data_najwa_shihab" yang juga baris data pertamanya diawali oleh indeks data ke-0.

Jadi alasan tersebutlah saya menggunakan parameter ignore_index = True di dalam fungsi pd.concat() untuk menghindari terjadinya duplikasi indeks data pada DataFrame yang bernama "comments_debate_df".

In [91]:
"""
Menampilkan jumlah baris data dan kolom yang dimiliki oleh variabel
DataFrame "comments_debate_df" yang merupakan DataFrame penggabungan dari
DataFrame "data_najwa_shihab" yang memiliki 22.999 data, DataFrame "data_kompastv" yang memiliki 12.500 data
dan terakhir adalah variabel DataFrame "data_kpu_ri" yang memiliki 12.740 data
"""
print("Jumlah Baris dan Kolom Data setelah Dilakukan Proses Penggabungan Ketiga Variabel DataFrame adalah:",
      comments_debate_df.shape)

Jumlah Baris dan Kolom Data setelah Dilakukan Proses Penggabungan Ketiga Variabel DataFrame adalah: (48239, 10)


In [93]:
# Mengeluarkan/menampilkan hanya nilai-nilai data (tanpa ada kolom) yang dimiliki oleh variabel DataFrame "comments_debate_df"
comments_debate_df.values

array([['Sbjsg3AFx00', 'UCo8h2TY_uBkAVUIc14m_KCA',
        'Ugx6mhUfDOUhd65fWyh4AaABAg', ...,
        'http://www.youtube.com/@udinlamidun9716',
        Timestamp('2024-01-07 17:03:53'),
        Timestamp('2024-01-09 23:18:49')],
       ['gUz_MgdwKg0', 'UCo8h2TY_uBkAVUIc14m_KCA',
        'UgxfMHgX25B2b3jkO4B4AaABAg', ...,
        'http://www.youtube.com/@zakishofi3442',
        Timestamp('2023-12-12 21:22:08'),
        Timestamp('2023-12-12 21:22:08')],
       ['tEbRB5xxOPI', 'UCo8h2TY_uBkAVUIc14m_KCA',
        'UgwTpSK1hjkTNQlcT894AaABAg', ...,
        'http://www.youtube.com/@sandigustiana',
        Timestamp('2024-01-21 21:32:13'),
        Timestamp('2024-01-21 21:32:13')],
       ...,
       ['yNO0YS846kU', 'UCVdSz86o9q2cQKvAQ5QhOYg',
        'UgxOwgAbtyVG412f0314AaABAg', ...,
        'http://www.youtube.com/@SpillProducInd',
        Timestamp('2023-12-13 12:38:48'),
        Timestamp('2023-12-13 12:38:48')],
       ['yNO0YS846kU', 'UCVdSz86o9q2cQKvAQ5QhOYg',
        'UgzJFZj2KoWhC

In [97]:
"""
Menampilkan top 5 nama user/akun YouTube yang paling sering berkomentar
serta teks komentar yang paling sering digunakan untuk berkomentar di dalam video debat capres-cawapres
Republik Indonesia tahun 2024 dari seluruh babak (babak pertama hingga kalima)
yang bersumber dari tiga channel YouTube yang berbeda yakni Najwa Shihab, KOMPASTV dan KPU RI
"""
print(comments_debate_df["author"].value_counts().head(), "\n")

print(comments_debate_df["text_comment"].value_counts().head())

@trianitarahmawati664        58
@rinideh6414                 56
@lasmiminah7171              52
@rzkadity69                  49
@dinarikaanggidesylia1408    49
Name: author, dtype: int64 

GASPOLL BRO! PRABOWO GIBRAN UNTUK INDONESIA MAJU    177
Amin                                                102
1                                                    83
2                                                    54
02                                                   47
Name: text_comment, dtype: int64
